spec/requests/admin_payroll_runs_spec.rb

```diff
@@ -0,0 +1,69 @@
+require "rails_helper"
+
+RSpec.describe "Admin payroll runs" do
+  context "when signed in as a service operator" do
+    let(:admin_session_id) { "some_user_id" }
+    before do
+      sign_in_to_admin_with_role(AdminSession::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE, admin_session_id)
+    end
+
+    describe "admin_payroll_runs#new" do
+      it "uses PayrollRun.payrollable_claims to generate the preview" do
+        claims = build_list(:claim, 3, :approved)
+        expect(PayrollRun).to receive(:payrollable_claims).and_return(claims)
```

Comment:
This couples the spec to the internal implementation detail of the controller and the model, making the test more brittle and also making it hard to refactor the code. Can we rewrite this spec to not stub our own code and instead verify something on the response instead? For a start, we should at least verify that we get a successful response. Next is probably verify that something that we'd expect to see if the code is working is present on the page.

Reply:
I've pushed an updated version of this spec, which removes the mocking. I'd like to ask some questions about it if you've got a moment today.

Reply:
I've pushed a simplified version of the spec, after our conversation.